### ISSUE TITLE
createOrUpdateRelease Nil pointer dereference

### ISSUE THREAD

**Issue by macox** [goreleaser/goreleaser#5130]
### What happened?

Upgrading from `1.13.1` to `2.2.0`

```
��� publishing
     ��� scm releases
       ��� releasing                                    tag=v0.0.2 repo=my-org/my-repo
       ��� could not check rate limits, hoping for the best...
       ��� could not check rate limits, hoping for the best...
       ��� took: 40s
   ��� took: 40s
 panic: runtime error: invalid memory address or nil pointer dereference
 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe28b72]
 
 goroutine 22 [running]:
 github.com/goreleaser/goreleaser/v2/internal/client.(*githubClient).createOrUpdateRelease(0xc00015c020, 0xc00022b108, 0xc0007e7278, {0xc0000605b0, 0x63})
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/client/github.go:454 +0x3b2
 github.com/goreleaser/goreleaser/v2/internal/client.(*githubClient).CreateRelease(0xc00015c020, 0xc00022b108, {0xc0000605b0, 0x63})
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/client/github.go:402 +0x339
 github.com/goreleaser/goreleaser/v2/internal/pipe/release.doPublish(0xc00022b108, {0x2ce2d40, 0xc00015c020})
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/pipe/release/release.go:127 +0x298
 github.com/goreleaser/goreleaser/v2/internal/pipe/release.Pipe.Publish({}, 0xc00022b108)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/pipe/release/release.go:106 +0x45
 github.com/goreleaser/goreleaser/v2/internal/pipe/publish.Pipe.Run.Handle.func1(0xc00005c678?)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/middleware/errhandler/error.go:15 +0x1b
 github.com/goreleaser/goreleaser/v2/internal/pipe/publish.Pipe.Run.PadLog.func2(0xc00022b108)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/middleware/logging/logging.go:46 +0x13d
 github.com/goreleaser/goreleaser/v2/internal/middleware/skip.Maybe.func1(0xc00022b108)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/middleware/skip/skip.go:42 +0x10c
 github.com/goreleaser/goreleaser/v2/internal/pipe/publish.Pipe.Run({{0xc00046e900?, 0xc0006cf350?, 0x4050220?}}, 0xc00022b108)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/pipe/publish/publish.go:84 +0x1a5
 github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1.Handle.1(0xc00005c510?)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/middleware/errhandler/error.go:15 +0x1b
 github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1.Log.2(0xc00022b108)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/middleware/logging/logging.go:30 +0x132
 github.com/goreleaser/goreleaser/v2/internal/middleware/skip.Maybe.func1(0xc00022b108)
 	github.com/goreleaser/goreleaser/v2@v2.2.0/internal/middleware/skip/skip.go:42 +0x10c
 github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1()
 	github.com/goreleaser/goreleaser/v2@v2.2.0/cmd/release.go:123 +0x182
 github.com/caarlos0/ctrlc.(*Ctrlc).Run.func1()
 	github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:48 +0x23
 created by github.com/caarlos0/ctrlc.(*Ctrlc).Run in goroutine 1
 	github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:47 +0x94
```

### How can we reproduce this?

`GO_VERSION="1.20" VERSION_PACKAGE="github.com/my-org/my-app/internal/version" goreleaser release --config=.goreleaser.yml`

```yaml
version: 2

env:
  - GO111MODULE=on
  - CGO_ENABLED=0
before:
  hooks:
    - go mod download

builds:
  - id: app-linux-amd64
    main: ./cmd/app/app.go
    binary: app

    ldflags:
      - -s -w -X "{{.Env.VERSION_PACKAGE}}.Version={{.Tag}}" -X "{{.Env.VERSION_PACKAGE}}.Commit={{.FullCommit}}" -X "{{.Env.VERSION_PACKAGE}}.BuildDate={{.Date}}" -X "{{.Env.VERSION_PACKAGE}}.GoVersion={{.Env.GO_VERSION}}"
    goos:
      - linux
    goarch:
      - amd64

archives:
  - name_template: "{{ .ProjectName }}-{{ .Os }}-{{ .Arch }}"
    files:
      - none*
      - extra/*

checksum:
  name_template: "{{ .ProjectName }}-checksums.txt"
  algorithm: sha256

changelog:
  disable: false

release:
  draft: false
  prerelease: false
  name_template: "{{.Tag}}"
```

### goreleaser version

```bash
v2.2.0
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
I think this was fixed in https://github.com/goreleaser/goreleaser/pull/5106

**Comment by macox** [comment]
> I think this was fixed in #5106

Ah sorry @caarlos0 I didn't spot that one. Do you think there will be a release containing the fix soon?

**Comment by caarlos0** [comment]
yes, sometime this week, maybe today even

gonna close this for now
